server/http/restapi/webhook.go

```diff
@@ -0,0 +1,68 @@
+package restapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/utils/httputils"
+)
+
+func (a *restapi) handleWebhook(w http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+
+	appID := vars["app_id"]
+	if appID == "" {
+		httputils.WriteBadRequestError(w, errors.New("app_id not specified"))
+		return
+	}
+
+	whName := vars["name"]
+	if whName == "" {
+		httputils.WriteBadRequestError(w, errors.New("webhook name not specified"))
+		return
+	}
+
+	queryVars := req.URL.Query()
+	if len(queryVars["secret"]) != 1 {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret was not provided"))
+		return
+	}
+
+	secret := queryVars["secret"][0]
+	if !a.validSecret(appID, secret) {
+		httputils.WriteBadRequestError(w, errors.New("webhook secret is not valid"))
+		return
+	}
+
+	var c interface{}
+	err := json.NewDecoder(req.Body).Decode(&c)
+	if err != nil {
+		httputils.WriteBadRequestError(w, err)
+		return
+	}
+
+	values := make(map[string]interface{})
+	values["payload"] = c
```

Comment:
We probably should add also the query values (minus the secret).

Reply:
We could do this:
1. read in req.Body as []byte
2. If content-type==application/json, parse it and inject it into "values" as such (AT THE TOP LEVEL, aka values = object)
3. pass the HTTP request values (all: URL, header) in the context (`Context.Request`?)

Reply:
Do you want me to create a new `Context.Request` of type `interface{}` and just fill it with the entire request?  Then the App would handle webhook the request however it wants?  That seem pretty flexible, and would remove any complexity placed on the proxy app.

Reply:
I am not sure whether it should go in the context or as a value. I feel that the Context.Request would only appear on webhooks, and therefore it does not feel so much as a call context, but as a value.

Reply:
All the HTTP stuff can go in as Values later, if needed (Query, Header)